### ISSUE TITLE
EntityManager / JPA

### ISSUE THREAD

**Issue by eepstein** [dropwizard/dropwizard#920]
I'm looking to use the general JPA APIs for persistence.  No need for any of the hibernate-specific functionality.  Is that something that already exists or is being contemplated?

**Comment by eepstein** [comment]
Is there any interest in this?  I've put together an initial impl. and am happy to contribute it as a bundle so that it may be improved by others.

**Comment by carlo-rtr** [comment]
Given we have JDBI & Hibernate, I don't think we need it.

The process for your getting your repo in the DW org is discussed here:
https://groups.google.com/d/msg/dropwizard-dev/eiiTDVFqmTc/K2gyDI90x7kJ

Alternatively, you can keep it in under your user name and we can list it here:
http://modules.dropwizard.io/thirdparty/

**Comment by silkentrance** [comment]
@eepstein any progress on the jpa bundle?

**Comment by eepstein** [comment]
I got it most of the way done. This was for DW 0.8.x however. 

With 0.9.2 the main need - transactions outside the persistence layer - was addressed. 

I can share what I have (if I can find it).

**Comment by silkentrance** [comment]
@eepstein how about making this a public project then :+1:

**Comment by eepstein** [comment]
Will do. I'll see where I put it. Was on a different dev box at the time.

**Comment by eepstein** [comment]
Hi Carsten,

I’m really sorry to say I don’t have access to the code anymore.  It was under contract at a company on their machines backed by their repo… they were OK with o/s contributions, but seems I never did that and since they’re not a current client I don’t have the code.  :-/

Ezra

**Comment by silkentrance** [comment]
@eepstein oh well, no problem. will look into this then myself...

**Comment by bladecatcher** [comment]
@silkentrance I'd put together JPA (for all persistence), Dropwizard using guice-persist (for DI and made writing transactional methods `@Transactional` annotation). The drawback is the @Transactional semantics don't work in Jersey 2 resources (as they're not loaded by Guice) for which I ended up having to use manual transaction management. An example implementation is presented here https://github.com/oregami/dropwizard-guice-jpa-seed .

Having said that, I don't recommend using guice with dropwizard at all, as it's painful to setup DI into Jersey resources and it can also break in the future due to incompatibilities.

**Comment by arteam** [comment]
Yes, it's a good advice. I think, If you are using JPA, it's just simplier to go with Spring, which has good integration with it. You can easily combine Spring and Dropwizard though.

**Comment by scottescue** [comment]
I just created and published a JPA module for Dropwizard that I derived from Dropwizard Hibernate.  It provides a shareable, thread-safe EntityManager that works with the UnitOfWork annotation.  There's no persistence.xml needed; setup and configuration is nearly identical to Dropwizard Hibernate. 

I hope other people find this useful.  So far I've published versions to work with Dropwizard 0.9.0 - 0.9.3.  I'm holding off on releasing a version compatible with Dropwizard 1.0.0 in hope I can get some feedback first.

__Github:__ https://github.com/scottescue/dropwizard-entitymanager
__User Guide:__ http://scottescue.com/dropwizard-entitymanager/
